### PR TITLE
feat: redeem

### PR DESCRIPTION
**Pull request by yuche**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `develop` branch(or to a previous version branch), _not_ the `master` branch
- [ ] All tests are passing
- [ ] Build success

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature

**Other information:**

### CODE REVIEW

**Review by ybian19 — APPROVED** [review]
verified

File: src/apis/ServerWalletAPI.ts [hunk 1]
```diff
@@ -444,6 +445,89 @@ export class ServerWalletAPI implements NFTWalletAPI {
     })
   }
 
+  async getAllRedeemEvents(page: number, type: RedeemListType) {
```
Comment (Vibes-INS) [inline src/apis/ServerWalletAPI.ts:448]: How about this?
```typescript
async getAllRedeemEvents(type: RedeemListType, options?: {
  page?: number
  limit?: number
}) {
  const params: Record<string, unknown> = {
    page: options?.page ?? 1,
    limit: options?.limit ?? PER_ITEM_LIMIT,
    type,
  }
  // ...
}
```

File: src/views/Reedem/Prize.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+          <div className="contain">{t('exchange.othder-prize')}</div>
+        </>
+      ) : null}
+      <div className="price-title">{prizes.reward_name}</div>
+      <div className="price-desc">{prizes.reward_description}</div>
+      <div className="imgs">
+        <PhotoProvider maskClassName="preview-mask" toolbarRender={() => null}>
+          {prizes.images.map((p) => {
+            return (
+              <div className="img">
```
Comment (Vibes-INS) [inline src/views/Reedem/Prize.tsx:175]: Do you need a key?
```jsx
<div className="img" key={i}>
```

**Review by Vibes-INS — APPROVED** [review]

File: src/apis/ServerWalletAPI.ts (hunk 1 above)
Comment (yuche) [reply src/apis/ServerWalletAPI.ts:448]: 感觉传 limit 进来没意义，这个是定死的，把他写到 env 都行

File: src/views/Reedem/Prize.tsx (hunk 2 above)
Comment (yuche) [reply src/views/Reedem/Prize.tsx:175]: 对，少了一个

File: src/i18n/en.json [hunk 3]
```diff
@@ -329,7 +331,126 @@
...
+        "held": "当前可兑换"
+      },
+      "warning": "兑换不会销毁您持有的 NFT，但会对其进行标记，标记过的 NFT 无法再次参与其他任何兑换活动。",
+      "warning-destroyed": "兑换将会销毁您参与兑换的 NFT ，该操作不可逆，一经销毁将无法找回",
+      "check": {
+        "price": "查看奖品",
+        "comment": "查看奖品和留言",
+        "wait": "• 待承兑"
+      },
+      "empty": "暂时还没有兑换",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:439]:
```suggestion
      "empty": "No record",
```

File: src/i18n/en.json [hunk 4]
```diff
@@ -329,7 +331,126 @@
...
+      "no-more": "没有更多兑换了",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:440]:
```suggestion
      "no-more": "No more",
```

File: src/i18n/en.json [hunk 5]
```diff
@@ -329,7 +331,126 @@
...
+      "error": "兑换失败，请检查当前网络情况或兑换条件",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:441]:
```suggestion
      "error": "Redeem failed, please check your network and redeem rules",
```

File: src/apis/ServerWalletAPI.ts (hunk 1 above)
Comment (Vibes-INS) [reply src/apis/ServerWalletAPI.ts:448]: 把limit作为一个可选项嘛，这样拓展性会比较好一点

Comment (yuche) [reply src/apis/ServerWalletAPI.ts:448]: 那应该把写成一个类型，所有 API 都这么搞。等你搞无线滚动组件的时候一起搞了把

Comment (Vibes-INS) [reply src/apis/ServerWalletAPI.ts:448]: 可以

File: src/i18n/en.json [hunk 6]
```diff
@@ -329,7 +331,126 @@
...
+      "result": {
+        "success": "兑换成功",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:443]:
```suggestion
        "success": "Redeem Successfully",
```

File: src/i18n/en.json [hunk 7]
```diff
@@ -329,7 +331,126 @@
...
+        "submited": "已提交兑换",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:444]:
```suggestion
        "submited": "Summitted",
```

File: src/i18n/en.json [hunk 8]
```diff
@@ -329,7 +331,126 @@
...
+        "time": "提交时间：",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:445]:
```suggestion
        "time": "Submit time: ",
```

File: src/i18n/en.json [hunk 9]
```diff
@@ -329,7 +331,126 @@
...
+        "fail": "兑换失败",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:446]:
```suggestion
        "fail": "Redeem failed",
```

File: src/i18n/en.json [hunk 10]
```diff
@@ -329,7 +331,126 @@
...
+        "custom": "活动发起人会根据您提供的接收信息进行承兑",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:447]:
```suggestion
        "custom": "The activity organizer will give out prize based on the info you provided",
```

File: src/i18n/en.json [hunk 11]
```diff
@@ -329,7 +331,126 @@
...
+        "check": "查看兑换的 NFT",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:448]:
```suggestion
        "check": "View Prizes",
```

File: src/i18n/en.json [hunk 12]
```diff
@@ -329,7 +331,126 @@
...
+        "confirm": "确认",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:449]:
```suggestion
        "confirm": "Confirm",
```

File: src/i18n/en.json [hunk 13]
```diff
@@ -329,7 +331,126 @@
...
+        "go-back": "返回兑换中心",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:450]:
```suggestion
        "go-back": "Back to redeem center",
```

File: src/i18n/en.json [hunk 14]
```diff
@@ -329,7 +331,126 @@
...
+        "error": "请检查当前网络情况或兑换条件",
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:451]:
```suggestion
        "error": "Please check your network and redeem rules",
```

File: src/i18n/en.json [hunk 15]
```diff
@@ -329,7 +331,126 @@
...
+        "redeeming": "兑换中..."
```
Comment (zhouyun-zoe) [inline src/i18n/en.json:452]:
```suggestion
        "redeeming": "Redeeming"
```

**Review by Vibes-INS — APPROVED** [review]

File: src/i18n/zh.json [hunk 16]
```diff
@@ -331,6 +333,125 @@
       "more": "更多",
       "no-on-sale": "无在售秘宝"
     },
+    "exchange": {
+      "title": "兑换中心",
+      "count": "{{count}} 个",
+      "redeemed": "已兑",
+      "nft-prize": "奖品包含下面所有 NFT",
+      "blind-prize": "奖品随机包含下面的 {{count}} 个",
```
Comment (zhouyun-zoe) [inline src/i18n/zh.json:341]:
```suggestion
      "blind-prize": "奖品随机包含下面的 {{count}} 个 NFT",
```

File: src/i18n/zh.json [hunk 17]
```diff
@@ -331,6 +333,125 @@
...
+        "held": "当前可兑换"
+      },
+      "warning": "兑换不会销毁您持有的 NFT，但会对其进行标记，标记过的 NFT 无法再次参与其他任何兑换活动。",
+      "warning-destroyed": "兑换将会销毁您参与兑换的 NFT ，该操作不可逆，一经销毁将无法找回",
+      "check": {
+        "price": "查看奖品",
+        "comment": "查看奖品和留言",
+        "wait": "• 待承兑"
+      },
+      "empty": "暂时还没有兑换",
```
Comment (zhouyun-zoe) [inline src/i18n/zh.json:439]:
```suggestion
      "empty": "无记录",
```

File: src/i18n/zh.json [hunk 18]
```diff
@@ -331,6 +333,125 @@
       "more": "更多",
       "no-on-sale": "无在售秘宝"
     },
+    "exchange": {
+      "title": "兑换中心",
+      "count": "{{count}} 个",
+      "redeemed": "已兑",
+      "nft-prize": "奖品包含下面所有 NFT",
+      "blind-prize": "奖品随机包含下面的 {{count}} 个",
+      "othder-prize": "发起人会在你提交兑换后根据你提供的接收信息进行承兑",
```
Comment (zhouyun-zoe) [inline src/i18n/zh.json:342]:
```suggestion
      "othder-prize": "发起人会在您提交兑换后根据您提供的接收信息进行承兑",
```

**Review by zhouyun-zoe — COMMENTED** [review]
update en translate

File: src/i18n/en.json [hunk 19]
```diff
@@ -329,7 +331,126 @@
...
+      "nft-prize": "奖品包含下面所有 NFT",
+      "blind-prize": "奖品随机包含下面的 {{count}} 个",
+      "othder-prize": "发起人会在你提交兑换后根据你提供的接收信息进行承兑",
+      "redeem-time": "兑换时间：",
+      "type": {
+        "token": "NFT",
+        "blind_box": "NFT 盲盒",
+        "custom": "其他"
+      },
+      "progress": "兑换进度",
```
[the remark this answers is not in the document]
Comment (yuche) [reply src/i18n/en.json:349]: 你这个应该是 Progress 把

**Pull request merged**